pandas/core/series.py

```diff
@@ -303,6 +293,23 @@ def _can_hold_na(self):
 
     _index = None
 
+    def _init_from_dict(self, data, index, dtype):
+        # Looking for NaN in dict doesn't work ({np.nan : 1}[float('nan')]
```

Comment:
can you make this a proper doc-string

Reply:
Just to double check: given the size of the method now, do you still prefer it isolated (rather than having exactly the same lines of code in ``__init__``)?

Reply:
yes absolutely, having these separate makes them much easier to grok.